### DISCUSSION TITLE
How to enable sample_rate for error logging

### DISCUSSION THREAD

**Discussion by Nikschavan** [stayallive/wp-sentry#202] [Q&A] [answered]
As per the docs, `sample_rate` can be set to limit the errors being reported to Sentry, but I did not find a way to set the settings in the API - https://docs.sentry.io/platforms/php/configuration/sampling/#sampling-error-events

**Comment by stayallive** [comment, marked as answer]
Best way currently is: https://github.com/stayallive/wp-sentry?tab=readme-ov-file#wp_sentry_options

This fires a little on the late side maybe depending on where your errors might occur.

Maybe [filtering](https://github.com/stayallive/wp-sentry?tab=readme-ov-file#wp_sentry_before_send) is a better way to prevent many errors, or can you share a bit more about your use case? I have personally never needed to sample for errors, but happy to add the option if it makes sense 👍

**Comment by Nikschavan** [comment]
Thank you. This should work for me.

My use case is that for the production environment, the same error is reported for all the traffic that consumes higher credits for the same error. 

So, I am planning to set a lower sample rate for production and 100% for all non-production environments.

**Reply by stayallive** [reply]
Okay, just be careful that the sample rate is totally random. So if you set the sample rate to 0.5 for example there is a 50% chance the error is reported so you might miss some errors because of this.
